app/assets/javascripts/components/articles/article_list.jsx

```diff
@@ -8,84 +9,101 @@ import List from '../common/list.jsx';
 import Article from './article.jsx';
 import CourseUtils from '../../utils/course_utils.js';
 
-const ArticleList = ({
-  articles,
-  course,
-  current_user,
-  actions,
-  articleDetails,
-  sortBy,
-  wikidataLabels,
-  sort
-}) => {
-  const keys = {
-    rating_num: {
-      label: I18n.t('articles.rating'),
-      desktop_only: true,
-      info_key: 'articles.rating_doc'
-    },
-    title: {
-      label: I18n.t('articles.title'),
-      desktop_only: false
-    },
-    character_sum: {
-      label: I18n.t('metrics.char_added'),
-      desktop_only: true,
-      info_key: 'articles.character_doc'
-    },
-    view_count: {
-      label: I18n.t('metrics.view'),
-      desktop_only: true,
-      info_key: 'articles.view_doc'
-    },
-    tools: {
-      label: I18n.t('articles.tools'),
-      desktop_only: false,
-      sortable: false
+const ArticleList = createReactClass({
+  displayName: 'ArticleList',
+
+  propTypes: {
+    articles: PropTypes.array,
+    course: PropTypes.object,
+    current_user: PropTypes.object,
+    actions: PropTypes.object,
+    articleDetails: PropTypes.object,
+    sortBy: PropTypes.func,
+    wikidataLabels: PropTypes.object,
+    sort: PropTypes.object
+  },
+
+  getInitialState() {
+    return {
+      selectedIndex: -1,
+    };
+  },
+
+  showDiff(index) {
+    this.setState({
+      selectedIndex: index
+    });
+  },
+
+  render() {
+    const keys = {
```

Comment:
No need for this constant to be defined inside `render`, where it will get redefined on every render. (I realize that's what's already happening in the stateless functional version of this component, but it's worth fixing now.) It can be constant outside of the class.

Reply:
Whoops, nevermind, looks like the `keys` object gets modified.